### PR TITLE
CEDS-1490 Add trait Injector which provide injector with clearing SharedMetrics

### PR DESCRIPTION
**Pull request by PatrykRudnicki**

### CODE REVIEW

File: test/base/WithInjector.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+package base
+
+import com.codahale.metrics.SharedMetricRegistries
+import play.api.inject.guice.GuiceApplicationBuilder
+
+trait WithInjector {
+
+  SharedMetricRegistries.clear()
```
Comment (ed0906) [inline test/base/WithInjector.scala]: Maybe add a comment too explain why this is here

File: test/base/WithInjector.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  val injector = GuiceApplicationBuilder().injector()
```
Comment (ed0906) [inline test/base/WithInjector.scala]: I think some helper methods could be useful here,

with `MockitoSugar` we get `mock[T]`, maybe a method that follows the same pattern e.g. `real[T]` or just `instanceOf[T]` so we can declutter the tests a little? What do you think?

File: test/connectors/CustomsDeclareExportsConnectorSpec.scala [hunk 3]
```diff
@@ -39,12 +37,9 @@ import unit.base.UnitSpec
 
 import scala.concurrent.ExecutionContext.global
 
-class CustomsDeclareExportsConnectorSpec extends UnitSpec with ScalaFutures {
+class CustomsDeclareExportsConnectorSpec extends UnitSpec with ScalaFutures with WithInjector {
```
Comment (ed0906) [inline test/connectors/CustomsDeclareExportsConnectorSpec.scala]: `with WithInjector` seems a little strange, why not just `with Injector`?

Or if that has name clashes maybe `with InjectorSugar` or something?

File: test/base/WithInjector.scala (hunk 2 above)
Comment (PatrykRudnicki) [reply test/base/WithInjector.scala]: Done, changes in separate commit :)

**Pull request merged**